api/v1alpha6/conversion.go

```diff
@@ -88,10 +88,36 @@ func restorev1alpha6MachineSpec(previous *OpenStackMachineSpec, dst *OpenStackMa
 	dst.Subnet = previous.Subnet
 }
 
+func restorev1alpha7MachineSpec(previous *infrav1.OpenStackMachineSpec, dst *infrav1.OpenStackMachineSpec) {
+	// PropagateUplinkStatus has been added in v1alpha7.
+	// We restore the whole Ports since they are anyway immutable.
+	dst.Ports = previous.Ports
+}
+
 func restorev1alpha7ClusterSpec(previous *infrav1.OpenStackClusterSpec, dst *infrav1.OpenStackClusterSpec) {
 	// APIServerLoadBalancer.Provider is new in v1alpha7
 	dst.APIServerLoadBalancer.Provider = previous.APIServerLoadBalancer.Provider
 	dst.Router = previous.Router
+	// PropagateUplinkStatus has been added in v1alpha7.
+	// We restore the whole Ports since they are anyway immutable.
+	if previous.Bastion != nil && previous.Bastion.Instance.Ports != nil {
+		dst.Bastion.Instance.Ports = previous.Bastion.Instance.Ports
+	}
+}
+
+func restorev1alpha7ClusterStatus(previous *infrav1.OpenStackClusterStatus, dst *infrav1.OpenStackClusterStatus) {
```

Comment:
If that was the case (v1alpha6 was storage version for v0.8), then I guess that would be true. But the storage version is set to v1alpha7, which makes much more sense IMO. So I'm thinking there would not be any conversions required "internally" and thus no need to restore it really (other than for happy fuzzing tests).

I was also thinking about what happens if the user asks for v1alpha6, makes changes to it and applies. Then it would be up-converted to v1alpha7, including the status. But the user should not normally make changes to the status (or even be able to) so maybe this scenario is irrelevant.

Reply:
As discussed in the office hours, let's keep this just in case. It should not do any harm
@mdbooth